substrate/file_utils

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef SUBSTRATE_FILE_UTILS
+#define SUBSTRATE_FILE_UTILS
+
+#include <algorithm>
+#include <iterator>
+#include <string>
+#include <vector>
+#include <substrate/fd>
+#include <substrate/mmap>
+
+namespace substrate
+{
+	template<typename Str = std::string>
+		std::vector<Str> read_lines(fd_t &file, typename Str::value_type separator = '\n') noexcept
+	{
+		using char_t = typename Str::value_type;
+
+		if (!file.valid())
+			return {};
+
+		const auto map{file.map(PROT_READ)};
+		if (!map.valid())
+			return {};
+
+		std::vector<Str> result;
+
+		const auto *const cbegin{map.address<char_t>()};
+
+		const auto *cend
+		{
+			[&]() {
+				const auto *begin {cbegin};
```

Comment:
*nods* Indeed, it just gets a little close to `const void *m_cvpThing;` for member naming and such which is why we like to avoid it. The updated diff looks good though, so marking as resolved.